nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -247,33 +233,33 @@ void Optimizer::generateNoisedTrajectories()
 void Optimizer::applyControlSequenceConstraints()
 {
   auto & s = settings_;
-
-  if (isHolonomic()) {
-    control_sequence_.vy = xt::clip(control_sequence_.vy, -s.constraints.vy, s.constraints.vy);
-  }
-
-  control_sequence_.vx = xt::clip(control_sequence_.vx, s.constraints.vx_min, s.constraints.vx_max);
-  control_sequence_.wz = xt::clip(control_sequence_.wz, -s.constraints.wz, s.constraints.wz);
-
-  float max_delta_vx = s.model_dt * s.constraints.ax_max;
-  float min_delta_vx = s.model_dt * s.constraints.ax_min;
-  float max_delta_vy = s.model_dt * s.constraints.ay_max;
-  float max_delta_wz = s.model_dt * s.constraints.az_max;
-  float vx_last = control_sequence_.vx(0);
-  float vy_last = control_sequence_.vy(0);
-  float wz_last = control_sequence_.wz(0);
-  for (unsigned int i = 1; i != control_sequence_.vx.shape(0); i++) {
+  float & vx_min = s.constraints.vx_min;
```

Comment:
A `float &` is actually larger than a `float` copy, since the reference is the same size as a pointer. In general, the copy is better, but I think you could just use `s.constraints.XYZ` inline below rather than reducing them into another variable altogether